packages/api/src/external/carequality/document/shared.ts

```diff
@@ -87,3 +92,60 @@ const generateCQFHIRContent = (
 
   return [cqFHIRContent];
 };
+
+function generateCQFHIRContained(authorInstitution: string | undefined): Organization | undefined {
+  if (!authorInstitution) return;
+
+  const org = splitNameAndOid(authorInstitution);
+  if (!org) return generateOrganization(authorInstitution);
+  const organization = generateOrganization(org.name, org.oid);
+
+  return organization;
+}
+
+export function generateOrganization(name?: string, oid?: string) {
+  const organization: Organization = {
+    resourceType: "Organization",
+  };
+  if (name) organization.name = name;
+  if (oid) {
+    organization.identifier = [
+      {
+        value: oid,
+      },
+    ];
+  }
+  return organization;
+}
+
+function splitNameAndOid(
+  input: string
+): { name: string | undefined; oid: string | undefined } | undefined {
+  const regex = /^(.+?)\^+(.+)$/;
```

Comment:
nit: keep this outside of the function so we don't need to create it on each time we call it.